sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -116,6 +119,7 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage, over
   override def applyModifier(mod: SidechainBlock): Try[SidechainState] = {
     validate(mod).flatMap { _ =>
       changes(mod).flatMap(cs => {
+        this.withdrawalEpochInfo = WithdrawalEpochUtils.getWithdrawalEpochInfo(mod, this.withdrawalEpochInfo, this.params)
```

Comment:
To avoid this operation here, BoxMinimalState trait can be reviewed. Even more, it seems that we can change it significantly or even remove.